admin_workspace_integration_test.go

```diff
@@ -14,6 +14,68 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestAdminWorkspaces_ListWithFilter(t *testing.T) {
+	client := testClient(t)
+	ctx := context.Background()
+
+	org, orgCleanup := createOrganization(t, client)
+	defer orgCleanup()
+
+	wTest1, wTest1Cleanup := createWorkspace(t, client, org)
+	defer wTest1Cleanup()
+
+	t.Run("when filtering workspaces on a current run status", func(t *testing.T) {
+		_, appliedCleanup := createRunApply(t, client, wTest1)
+		t.Cleanup(appliedCleanup)
+
+		wl, err := client.Admin.Workspaces.List(ctx, &AdminWorkspaceListOptions{
+			Filter: "applied", Include: []AdminWorkspaceIncludeOpt{AdminWorkspaceCurrentRun},
+		})
+
+		require.NoError(t, err)
+		require.NotEmpty(t, wl.Items)
+	})
```

Comment:
❓Looking for weigh-in: since this API call with an admin token will get all workspaces across all orgs, slicing at `Items[0]` to check for a run status applied won't exactly work. And even though this workspace applies the run, the `CurrentRun` attr was `null` when I printed the output of wTest1 obj.

Reply:
For this given test: `filtering workspaces on current run status`, if you filter on status `applied` each workspace in the returned list should have a current run with status `applied`.

Reply:
I went ahead and pushed an assertion on the run status, but some things I am thinking about:

- when this API call returns several workspaces across different orgs, then how do we ensure the correct workspace run status is accounted for?
- `Items` is an array of objs, so this tests passes because `Items[0]` is the only object with a `CurrentRun.Status`

Reply:
> when this API call returns several workspaces across different orgs, then how do we ensure the correct workspace run status is accounted for?

You're already doing so by:
```go
for _, wkspace := range wl.Items {
  assert.Equal(t, wkspace.CurrentRun.Status, "the-status-you-filtered-on")
}
``` 
Furthermore, I would use the type `RunStatus` for the `Filter` param instead of `string`. That way you can perform validations and instead assert:
```go
assert.Equal(t, wkspace.CurrentRun.Status, RunApplied)
```


> Items is an array of objs, so this test passes because Items[0] is the only object with a CurrentRun.Status

Correct, you may have one workspace in the returned list because the tests have only created **one** applied run on **one** workspace so far (I wouldn't be confident that this was always the case however). If you want to feel more confident that it's working properly, create several applied runs.